[milestone/12.1] Milestone 12.1

felix-schwarz opened the pull request:
## Description
Implements these changes:
- new feature: adds new `last used` sort method
- enhancement: turn `Available Offline` / `Make unavailable offline` pair of actions into a single `Available Offline` action with a switch
- enhancement: add modern password manager integration to `BookmarkViewController` via input field content types
- enhancement: move metrics logging to response logging block
- fix: `Quick Access > Recents` is sorted by `last used` date

## Related Issues
- implemented: #1258 
- fixed: #1243

## Screenshots (if appropriate):

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

felix-schwarz committed:
- CreateDocumentAction: make availability dependant on OCAppProvider.supportsCreateDocument
- AccountConnection: create Open with actions dependant on OCAppProvider.supportsCreateDocument
- OpenInWebAppAction:
	- introduce new "auto" mode that uses the best available method - or shows an error
	- limit size of icons to a maximum of 32x32 points
- update SDK for new feature detection capabilities

felix-schwarz committed:
- add support for TUS upload mtime metadata header (fixing #1227)

felix-schwarz committed:
- FileProviderExtension: add additional debug output for enumerator

felix-schwarz committed:
FileProvider:
- add missing errorHandler calls to FileProviderContentEnumerator
- add last resort enumeration completion call to FileProviderEnumeratorObserver.dealloc
- add additional debug logging

felix-schwarz committed:
- FIleProviderContentEnumerator: add/refine more debug logging

felix-schwarz committed:
- bump version to 12.0.3 and build number to 271

felix-schwarz committed:
- update SDK to fix https://github.com/owncloud/enterprise/issues/5811

felix-schwarz committed:
- OCSavedSearch+Interactions: add attribute .useSortDescriptor and utility method useSortDescriptor() to OCSavedSearch
- AccountController: add sort descriptor override to "Recents" item
- SortMethod: add new "lastUsed" sort method that sorts by .lastUsed and .lastModified

felix-schwarz committed:
- SortBar:
	- replace SortDirection/SortMethod with SortDescriptor for full consistency
	- update SortBarDelegate to no longer set the delegate's variables directly
	- update SortBarDelegate methods to use SortDescriptor
- ClientItemViewController:
	- fully embrace SortDescriptor
	- replace SortDirection/SortMethod ivar with a single SortDescriptor ivar
	- move read/write of default sort descriptor to SortDescriptor.defaultSortDescriptor
	- allow override of initial SortDescriptor via incoming clientContext.sortDescriptor
	- ensure that the default sort descriptor is only updated on actual user changes

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/owncloud/ios-app?pullRequest=1257) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/owncloud/ios-app?pullRequest=1257) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/owncloud/ios-app?pullRequest=1257) it.</sub>

felix-schwarz committed:
- OCCore+Extension: new convenience method to update the .lastUsed date of an item
- OpenInWebAction: update last used date when opening an item via a web app
- add missing translation strings for "last used" for English and German

felix-schwarz committed:
- replace AvailableOfflineAction and MakeAvailableOfflineAction with AvailableOfflineAction and a switch

felix-schwarz committed:
- change name of AvailableOfflineAction from "Available offline" to "Make available offline"

felix-schwarz committed:
Configuration documentation updated

felix-schwarz committed:
- AvailableOfflineAction: only show action for items not already available offline via AvailableOffline policy on parent item

felix-schwarz committed:
- AvailableOfflineAction: only show action for items not already available offline via AvailableOffline policy on parent item

felix-schwarz committed:
Merge branch 'master' into milestone/12.1

# Conflicts:
#	ios-sdk
#	ownCloud File Provider/FileProviderContentEnumerator.m
#	ownCloud/Resources/de.lproj/Localizable.strings
#	ownCloudAppShared/Client/Actions/Implementations/OpenInWebAppAction.swift

felix-schwarz committed:
Merge from milestone/12.0.4:
- BookmarkViewController: add modern password manager integration to BookmarkViewController via input field content types
- SDK update: OCHTTPPipeline: move metrics logging to response logging block

felix-schwarz committed:
Merge branch 'master' into milestone/12.1

felix-schwarz committed:
Calens changelog updated
